src/mscorlib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -240,38 +240,58 @@ private static IEnumerable<string> GetTimeZoneIds(string timeZoneDirectory)
             catch (IOException) { }
             catch (UnauthorizedAccessException) { }
 
-            List<string> timeZoneIds = new List<string>();
-            if (zoneTabFileLines != null)
+            if (zoneTabFileLines == null)
             {
-                foreach (string zoneTabFileLine in zoneTabFileLines)
+                return new List<string>();
+            }
+
+            // Estimate the number of ids in the file based on the total number of lines
+            // minus the number of comment lines at the beginning of the file (the tab file
+            // typically only has comment lines at the beginning of the file).
+            int estimatedTimeZoneIdCount = zoneTabFileLines.Length;
+            foreach (string zoneTabFileLine in zoneTabFileLines)
+            {
+                if (!string.IsNullOrEmpty(zoneTabFileLine) && zoneTabFileLine[0] == '#')
```

Comment:
I am not sure whether this optimization to estimate the number of ids is worth the extra code. Using the number of lines as the initial list capacity should be almost as good, and I guess it is likely going to disappear anyway if you switch to more low-level parser.